jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -44,6 +44,15 @@
 JERRY_STATIC_ASSERT ((uint32_t) ((int32_t) ECMA_STRING_MAX_CONCATENATION_LENGTH) ==
                      ECMA_STRING_MAX_CONCATENATION_LENGTH);
 
+/**
+ * String header
+ */
+typedef struct ecma_string_heap_header_t
+{
+  uint16_t size;
+  uint16_t length;
+} ecma_string_heap_header_t;
```

Comment:
@LaszloLango That does not seem valid C to me. `typedef struct { ... } xxx_t;` could work however.